Libraries/SDHC/ff13/Source/diskio.c

```diff
@@ -146,33 +300,98 @@ DRESULT disk_ioctl (
 	void *buff		/* Buffer to send/receive control data */
 )
 {
-    DRESULT status;
+	if(!flash){
+
+		
+    	DRESULT status;
+
+    	switch(cmd) {
+    	    case CTRL_SYNC:
+			    /* Mandatory */
+		    	status = ctrl_sync(buff);
+		    	break;
+    	    case GET_SECTOR_COUNT:
+			    /* Mandatory */
+			    status = get_sector_count(buff);
+			    break;
+    	    case GET_BLOCK_SIZE:
+			    /* Mandatory */
+			    status = get_block_size(buff);
+			    break;
+    	    case MMC_GET_CSD:
+			    /* Optional */
+			    status = mmc_get_csd(buff);
+			    break;
+			default:
+			    status = RES_PARERR;
+			    break;
+    	}
+	
+    	return status;
+	}
+#ifdef FLASH
+	else{
+		    DRESULT status;
 
+
+    status = RES_OK;
+#if FF_MAX_SS == 4096
```

Comment:
Same `FF_MAX_SS` considerations here - please add a comment explaining why it's necessary and consider consolidating the switch statements if possible